### PR TITLE
Update DCHDigi for using correct wire position from DDSegmentation::GridDriftChamber

### LINKED ISSUE #62: WIP: Get the DCH cell position using DD4HEP in DCHDigi
To do: Get the DCH cell position using DD4HEP in DCHDigi

### PR DESCRIPTION
**Pull request by wenxingfang**
Update DCHDigi for using correct wire position from DDSegmentation::GridDriftChamber

Fixes #62 

image

### CODE REVIEW

File: Digitisers/DCHDigi/src/DCHDigiAlg.cpp [hunk 1]
```diff
@@ -29,39 +29,83 @@ DCHDigiAlg::DCHDigiAlg(const std::string& name, ISvcLocator* svcLoc)
...
+  m_segmentation = dynamic_cast<dd4hep::DDSegmentation::GridDriftChamber*>(readout.segmentation().segmentation());
+  m_decoder = m_geosvc->getDecoder(m_readout_name);
+  if (!m_decoder) {
+      error() << "Failed to get the decoder. " << endmsg;
+      return StatusCode::FAILURE;
+  }
+
+  std::string s_output=m_Output; 
+  if(m_WriteAna){
+      m_fout = new TFile(s_output.c_str(),"RECREATE");
```
Comment (mirguest) [inline Digitisers/DCHDigi/src/DCHDigiAlg.cpp:52]: Hi @wenxingfang , 
Could you use Gaudi's NtupleSvc?
Tao

Comment (wenxingfang) [reply Digitisers/DCHDigi/src/DCHDigiAlg.cpp:52]: Hi @mirguest 
I want to use std::vector<> for branch variable. Does Gaudi's NtupleSvc has similar function ?

Best,
Wenxing

Comment (mirguest) [reply Digitisers/DCHDigi/src/DCHDigiAlg.cpp:52]: Sure, please have a look at https://gitlab.cern.ch/gaudi/Gaudi/-/tree/master/GaudiExamples/src/NTuples
Tao

Comment (wenxingfang) [reply Digitisers/DCHDigi/src/DCHDigiAlg.cpp]: Hi @mirguest Thanks! I have updated it, please check it!

Best,
Wenxing

**Review by mirguest — APPROVED** [review]
Looks good. Thank you.
Tao

**Pull request merged**